src/sheet_metal_conus.py

```diff
@@ -126,31 +132,69 @@ def makeMarkerstyle(self, name, rotate):
             self.arrowlen = 6.0
-            self.dimline_style['marker-start'] = 'url(#ArrowDIN-start)'
-            self.dimline_style['marker-end'] = 'url(#ArrowDIN-end)'
+            style['marker-start'] = 'url(#ArrowDIN-start)'
+            style['marker-end'] = 'url(#ArrowDIN-end)'
             self.makeMarkerstyle('ArrowDIN-start', False)
             self.makeMarkerstyle('ArrowDIN-end', True)
         else:
             # outside
             self.arrowlen = 0
-            self.dimline_style['marker-start'] = 'url(#ArrowDINout-start)'
-            self.dimline_style['marker-end'] = 'url(#ArrowDINout-end)'
+            style['marker-start'] = 'url(#ArrowDINout-start)'
+            style['marker-end'] = 'url(#ArrowDINout-end)'
             self.makeMarkerstyle('ArrowDINout-start', False)
             self.makeMarkerstyle('ArrowDINout-end', True)
 
-    
+    def drawDimArc(self, center, start, end, radius, style, parent, gap=0, lowside=True):
+        " just the arrowed arc line "
+        angle = abs(end-start)
+        # inside or outside 
+        inside = True
+        critical_length = 35
+        dist = calc_dist_between_points(point_on_circle(radius, start), point_on_circle(radius, end))
+        if angle < 45 and dist > critical_length: inside = False
+        # change start and end angles to make room for arrow markers
+        arrow_angle = math.degrees(math.sin(self.arrowlen/radius))
+        if lowside:
+            start += arrow_angle
+            angle -= arrow_angle
+            anglefac = 1
+        else:
+            start -= arrow_angle
+            angle -= arrow_angle
+            anglefac = -1
+        #
+        if gap == 0:
+            line_attribs = {'style' : simplestyle.formatStyle(style),
+                            'd'     : self.build_arc(center, start, angle*anglefac, radius, lowside) }
+            ell = inkex.etree.SubElement(parent, inkex.addNS('path','svg'), line_attribs )
+        else: # leave a gap for label
+            gap_angle = math.degrees(math.sin(gap/radius))
+            startstyle = deepcopy(style)
+            startstyle['marker-start'] = None
+            line_attribs = {'style' : simplestyle.formatStyle(startstyle),
+                            'd'     : self.build_arc(center, start, angle*anglefac/2-gap_angle/2*anglefac, radius, lowside) }
+            ell = inkex.etree.SubElement(parent, inkex.addNS('path','svg'), line_attribs )
+            endstyle = deepcopy(style)
+            endstyle['marker-end'] = None
+            line_attribs = {'style' : simplestyle.formatStyle(endstyle),
+                            'd'     : self.build_arc(center, angle/2*anglefac+gap_angle/2*anglefac, angle*anglefac, radius, lowside) }
+            inkex.etree.SubElement(parent, inkex.addNS('path','svg'), line_attribs )
+        # return pos in center of gap (or arc)
+        textposangle = angle/2*anglefac
+        return (point_on_circle(radius, math.radians(textposangle)))
+        
```

Comment:
arc drawing code which can split an arc into two, leaving a gap.
Also calculates where to start and end arc so arrow markers will fit correctly.